back/app/models/initiative.rb

```diff
@@ -145,6 +143,16 @@ def review_status?
 
   private
 
+  def proposed_at
+    initiative_status_changed_at('proposed')
+  end
+
+  def initiative_status_changed_at(initiative_status_code)
+    initiative_status_changes
+      .where(initiative_status: InitiativeStatus.where(code: initiative_status_code))
+      .order(:created_at).pluck(:created_at).last
+  end
```

Comment:
This is a smart solution! I totally overlooked the fact that we have the date of each status change saved already in the `intitative_status_changes` records!